aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/CreateHandler.java

```diff
@@ -55,4 +69,14 @@
             throw new CfnInvalidRequestException(ResourceModel.TYPE_NAME + e.getMessage(), e);
         }
     }
+
+    private static List<String> extractPrincipalsForAgentPermissions(final ResourceModel model) {
+        if (model.getPermissions() == null) {
+            return emptyList();
+        }
+        if (model.getPermissions().getAgentPermissions() == null) {
```

Comment:
Could you check whether it is possible to have Permission == null / AgentPermissions == null before reaching this line?

I think, if any of those field are empty/ null, you would get Invalid request exception anyway: 

```
Invalid request provided: AWS::CodeGuruProfiler::ProfilingGroup1 validation error detected: Value null at 'principals' failed to satisfy constraint: Member must not be null
```

Reply:
These checks for null are to to avoid a NullPointerException when returning the principals. For now, it looks like the PutPermissions API requires a minimum of 1 principal, but I would prefer to let the code rely on that API validation, and not to duplicate it.

* https://docs.aws.amazon.com/codeguru/latest/profiler-api/API_PutPermission.html#profiler-PutPermission-request-principals

Reply:
Actually, considering also Ivo's comment about ```Permissions``` not being required though, I will update the code as suggested, to not make the call if no principals are set. Thanks.